### PR TITLE
manifests/gcp: Added Network Project ID to the manifest

### PR DESCRIPTION
**Pull request by barbacbd**
** Added the network project ID to match that of
https://github.com/openshift/origin/blob/57b52e0a9528027448422d678c4ef682dfc0b885/vendor/k8s.io/legacy-cloud-providers/gce/gce.go#L136

[CORS-2037](https://issues.redhat.com//browse/CORS-2037)

### CODE REVIEW

**Comment by barbacbd** [comment]
requires #6166

File: pkg/asset/manifests/gcp/cloudproviderconfig.go [hunk 1]
```diff
@@ -61,5 +66,6 @@ node-tags       = {{$tag}}
 node-instance-prefix = {{.Global.NodeInstancePrefix}}
 external-instance-groups-prefix = {{.Global.ExternalInstanceGroupsPrefix}}
 subnetwork-name = {{.Global.SubnetworkName}}
+network-project-id = {{.Global.NetworkProjectID}}
```
Comment (jstuever) [inline pkg/asset/manifests/gcp/cloudproviderconfig.go:69]: What are the implications of always passing network-project-id here? Should we put in some logic to only include it when network-project-id is not empty?

File: pkg/asset/manifests/gcp/cloudproviderconfig.go [hunk 2]
```diff
@@ -51,7 +56,7 @@ func CloudProviderConfig(infraID, projectID, subnet string) (string, error) {
 	return buf.String(), nil
 }
 
-var configTmpl = `[global]
+var configTmpl = fmt.Sprintf(`[global]
```
Comment (patrickdillon) [inline pkg/asset/manifests/gcp/cloudproviderconfig.go:59]: why wrap this with `fmt.Sprintf`? It seems like it is not needed.

Comment (barbacbd) [reply pkg/asset/manifests/gcp/cloudproviderconfig.go:59]: Oh that is a mistake that snuck back in

**Comment by patrickdillon** [comment]
/approve

LGTM with a successful GCP test run

**Pull request merged**